source/framework/vx_reference.c

```diff
@@ -401,19 +401,23 @@ vx_status ownCopyReferenceGeneric(vx_reference input, vx_reference output)
     tivx_shared_mem_ptr_t  *op_mem_ptr = NULL;
     uint32_t                op_mem_size = 0;
     vx_status status;
-
-    if(((vx_bool)vx_true_e == ownIsGenericAllocReferenceType(input->type)) &&
-       ((vx_bool)vx_true_e == ownIsGenericAllocReferenceType(output->type)))
+     /* added void here as this status check of
+    * ownReferenceGetMemAttrsFromObjDesc will always be true if
+    * the previous condition is true
+    */
+    (void)ownReferenceGetMemAttrsFromObjDesc(input, &ip_mem_ptr, &ip_mem_size);
+    (void)ownReferenceGetMemAttrsFromObjDesc(output, &op_mem_ptr, &op_mem_size);
+    status = ownReferenceLock(output);
+    if ((vx_status)VX_SUCCESS == status)
     {
-        /* added void here as this status check of
-        * ownReferenceGetMemAttrsFromObjDesc will always be true if
-        * the previous condition is true
-        */
-        (void)ownReferenceGetMemAttrsFromObjDesc(input, &ip_mem_ptr, &ip_mem_size);
-        (void)ownReferenceGetMemAttrsFromObjDesc(output, &op_mem_ptr, &op_mem_size);
-        status = ownReferenceLock(output);
-        if (((vx_status)VX_SUCCESS == status) &&
-            (ip_mem_size <= op_mem_size))
+#ifdef LDRA_UNTESTABLE_CODE
```

Comment:
A test for this has been added for a vx_image (testImportFromHandle) where vxCreateImageFromHandle() is used to get an image from memory not allocated with openVX. For Tensor, we don't have the corresponding function yet so that is why the exception is still there.